recent regression adding http headers to push/fetch

### Reproduction steps

I use a bearer token in an Authorization header to authorize with my server.

Using git2go:

```go
opts := new(git.PushOptions)
opts.Headers = []string{"Authorization: Bearer " + bearer}
err := remote.Push(refspecs, opts)
```

This worked until recently. At current master (559751714b23f63ead726e9a773be8b07e3551ae), there is no Authorization header present when the request reaches the server.

I have bisected to one of these three commits:

* 7372573b5f1113b8522e2588fac1c529ddcedb0a
* bf55facf15e5b6897ff7305162a77d92089f6c82
* b9c5b15a7958ab4ecb83504a6d858efd18610297

The first two of these commits don't compile: `src/transports/http.c:89:2: error: unknown type name 'git_http_authtype_t'`. The last one seems pretty plausible.

cc @ethomson

--- a/src/transports/http.c
+++ b/src/transports/http.c
@@ -346,6 +346,7 @@ static int generate_request(
 	request->credentials = transport->server.cred;
 	request->proxy = use_proxy ? &transport->proxy.url : NULL;
 	request->proxy_credentials = transport->proxy.cred;
+	request->custom_headers = &transport->owner->custom_headers;
 
 	if (stream->service->method == GIT_HTTP_METHOD_POST) {
 		request->chunked = stream->service->chunked;

--- a/tests/online/clone.c
+++ b/tests/online/clone.c
@@ -392,6 +392,21 @@ void test_online_clone__credentials(void)
 	cl_fixture_cleanup("./foo");
 }
 
+void test_online_clone__credentials_via_custom_headers(void)
+{
+	const char *creds = "libgit3:libgit3";
+	git_buf auth = GIT_BUF_INIT;
+
+	cl_git_pass(git_buf_puts(&auth, "Authorization: Basic "));
+	cl_git_pass(git_buf_encode_base64(&auth, creds, strlen(creds)));
+	g_options.fetch_opts.custom_headers.count = 1;
+	g_options.fetch_opts.custom_headers.strings = &auth.ptr;
+
+	cl_git_pass(git_clone(&g_repo, "https://bitbucket.org/libgit2/testgitrepository.git", "./foo", &g_options));
+
+	git_buf_dispose(&auth);
+}
+
 void test_online_clone__bitbucket_style(void)
 {
 	git_credential_userpass_payload user_pass = {

